### PR TITLE
Remove hardcode emails | Defer action | remove session accessToken

### PR DESCRIPTION
**Pull request by GabrielPedroza**
## Description

PR Accomplishment: Emails are now not hardcoded and able to be executed with delay if user chooses it so. remove session access token because of initial error it causes (breaks production for brand new users)

Future PR Accomplishment: Refer to https://github.com/GabrielPedroza/bread/pull/24

## Milestones

THIRD TECHNICAL CHALLENGE DONE!!! (defer execution)

## Test Plan

Test Plan for remove hardcode emails: Refer to PR https://github.com/GabrielPedroza/bread/pull/23. New input fields so check images below.

Reason for removal of session access token: Refer to PR https://github.com/GabrielPedroza/bread/pull/11 for creation. This explanation has a bug. When the user first logs in with their GitHub account (as a brand new user), an error will occur: https://pastebin.com/jfNdUtNG
Test Plan for remove session access token: This question was asked in Web Tech Q&A. Access Token still works as intended but will expire after some time. Should not impede with demo

Test Plan for Defer action: Refer to PR https://github.com/GabrielPedroza/bread/pull/12 for how to create an automation. New Ruleset modal forms will be shown in the UI below. All fields are required except for delayed email (in minutes) input. If not entered, it will be zero and it is not possible to enter a number lower than 0. For testing purposes, input 1 or 2 minutes. This implementation also works if server restarts as it is stored as a scheduled email elsewhere from the project inside the API from SendGrid.

New Form input fields (including defer execution): https://ibb.co/SPK6ZgP

### CODE REVIEW

File: src/components/FormType.tsx [hunk 1]
```diff
@@ -79,7 +84,7 @@ const GitHubFormType = ({ modals }: GitHubFormTypeProps) => {
 
       // passing accessToken is necessary because DB access token can be stale which can cause 401: Bad Credentials
       const createWebhookResultObject = await createWebhook.mutateAsync({
-        accessToken: session.user.accessToken,
+        // accessToken: session.user.accessToken,
```
Comment (yangdanny97) [inline src/components/FormType.tsx:87]: why is this commented out, and what does that mean for the `// passing accessToken is necessary...`  comment above

File: src/server/auth.ts [hunk 2]
```diff
@@ -1,12 +1,11 @@
 import { PrismaAdapter } from "@next-auth/prisma-adapter";
 import { type GetServerSidePropsContext } from "next";
 import GitHubProvider from "next-auth/providers/github";
-import GoogleProvider from "next-auth/providers/google";
+// import GoogleProvider from "next-auth/providers/google";
```
Comment (yangdanny97) [inline src/server/auth.ts:4]: ?

File: src/server/auth.ts [hunk 3]
```diff
@@ -34,13 +33,13 @@ declare module "next-auth" {
   }
 }
 
-interface UserWithAccessToken extends User {
-  tokens: {
-    access_token: string;
-  };
-}
+// interface UserWithAccessToken extends User {
```
Comment (yangdanny97) [inline src/server/auth.ts:36]: ?

File: src/server/auth.ts [hunk 4]
```diff
@@ -70,18 +69,19 @@ export const authOptions: NextAuthOptions = {
...
-          /* eslint-disable-next-line @typescript-eslint/no-unsafe-assignment */
-          profileWithAccessToken = { ...profile, tokens };
-        }
-        return profileWithAccessToken;
-      },
-    }),
-    GoogleProvider({
-      clientId: env.GOOGLE_CLIENT,
-      clientSecret: env.GOOGLE_SECRET,
+      // DANGEROUS SESSION ACCESS TOKEN - Errors when user first attempts to log in the site but works afterwards.
```
Comment (yangdanny97) [inline src/server/auth.ts:72]: ?

**Comment by yangdanny97** [comment]
lots of commented out code here

File: src/components/FormType.tsx (hunk 1 above)
Comment (GabrielPedroza) [reply src/components/FormType.tsx:87]: Reason is in test plan in pr template

**Comment by GabrielPedroza** [comment]
> lots of commented out code here

https://github.com/GabrielPedroza/bread/pull/25#discussion_r1285377144

**Pull request merged**